docs/user-guide/rug/generators.md

```diff
@@ -129,10 +202,59 @@ highly recommended.
 We define the class that will implement our generator on line 8.  A
 generator implements the `#!typescript PopulateProject` interface.
 This interface requires the `#!typescript populate(Project)` method to
-be defined, which we do on line 20 (more on that below).  It is
+be defined, which we do on line 10 (more on that below).  It is
 convention for the generator and the class that implements it to have
 the same name.
 
+In the last line of the generator script we export an instance of that
+generator to make it visible to the Rug runtime when it executed (line 15).
+Like the generator class name, the name of the `#!typescript const`
+does not matter, but it is convention to use the generator/class name,
+lower-casing the first letter.
+
+At this stage, the generator does not do anything useful. As explained earlier,
+a generator copies the content of the project where it lives into a target 
+directory before applying changes. Let's now amend the generator to change the 
+name of the copied class. An action users would likely do manually.
+
+```typescript linenums="1"
+import { PopulateProject } from '@atomist/rug/operations/ProjectGenerator'
+import { Project } from '@atomist/rug/model/Project'
+import { File } from '@atomist/rug/model/File'
+import { Pattern } from '@atomist/rug/operations/RugOperation'
+import { Generator, Parameter, Tags } from '@atomist/rug/operations/Decorators'
+
+@Generator("NewSpringBootService", "Rug project for Spring Rest Services")
+@Tags("documentation")
+class NewSpringBootService implements PopulateProject {
+
+    @Parameter({
+        displayName: "Class Name",
+        description: "name for the service class",
+        pattern: Pattern.java_class,
+        validInput: "a valid Java class name, which contains only alphanumeric characters, $ and _ and does not start with a number",
+        minLength: 1,
+        maxLength: 50,
+        required: false
+    })
+    service_class_name: string;
+
+    populate(project: Project) {
+        console.log(`Creating ${project.name()}`);
+
+        project.findFile("src/main/java/com/company/MyRestServiceApplication.java").replace("MyRestService", this.service_class_name)
```

Comment:
Semicolons.

We should probably be doing something like https://github.com/atomist-rugs/common-editors/blob/master/.atomist/editors/ClassRenamer.rug#L31-L32 rather than brute force replace.